crates/prune/prune/src/segments/user/history.rs

```diff
@@ -151,7 +151,7 @@ where
         } else {
             cursor.upsert(
                 RawKey::new(key),
-                RawValue::new(BlockNumberList::new_pre_sorted(higher_blocks)),
+                RawValue::new(BlockNumberList::new_pre_sorted(higher_blocks())),
```

Comment:
why do this and call `higher_blocks` two times?

Reply:
iterator vs collecting

Reply:
`blocks` can be up to 2000, is iterating over it two times faster than allocating once?

Reply:
Probably not